lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2789,8 +2789,12 @@ class VJPEmitter final
   static SubstitutionMap getSubstitutionMap(SILFunction *original,
                                             SILFunction *vjp) {
     auto substMap = original->getForwardingSubstitutionMap();
-    if (auto *vjpGenEnv = vjp->getGenericEnvironment())
-      substMap = substMap.subst(vjpGenEnv->getForwardingSubstitutionMap());
+    if (auto *vjpGenEnv = vjp->getGenericEnvironment()) {
+      auto vjpSubstMap = vjpGenEnv->getForwardingSubstitutionMap();
+      substMap = SubstitutionMap::get(
+          vjpGenEnv->getGenericSignature(), QuerySubstitutionMap{vjpSubstMap},
+          LookUpConformanceInSubstitutionMap(vjpSubstMap));
```

Comment:
cc @slavapestov: is this the optimal way to get the desired substitution map?

---

For reference (bottommost substitution map is the goal):
```
`original->getForwardingSubstitutionMap()`:
(substitution_map generic_signature=<Scalar where Scalar : Numeric>
  (substitution Scalar -> Scalar)
  (conformance type=Scalar
    (abstract_conformance protocol=Numeric)))

`vjp->getForwardingSubstitutionMap()`:
(substitution_map generic_signature=<τ_0_0 where τ_0_0 : FloatingPoint>
  (substitution τ_0_0 -> τ_0_0)
  (conformance type=τ_0_0
    (abstract_conformance protocol=FloatingPoint)))

---

`VJPEmitter.SubsMap` BEFORE this patch:
(substitution_map generic_signature=<τ_0_0 where τ_0_0 : Numeric>
  (substitution τ_0_0 -> τ_0_0)
  (conformance type=τ_0_0
    (abstract_conformance protocol=Numeric)))

`VJPEmitter.SubsMap` AFTER this patch (goal):
(substitution_map generic_signature=<τ_0_0 where τ_0_0 : FloatingPoint>
  (substitution τ_0_0 -> τ_0_0)
  (conformance type=τ_0_0
    (abstract_conformance protocol=FloatingPoint)))
```